### ISSUE TITLE
strict-raw-types not working while constrained

### ISSUE THREAD

**Issue by JCKodel** [dart-lang/linter#4440]
**Describe the issue**

While using this analysis_options.yaml:

```yaml
analyzer:
  language:
    strict-casts: true
    strict-raw-types: true
    strict-inference: true
```

This example only triggers a warning in the first usage (while it should trigger for both):

```dart
class AnotherClass {}

class Something {
  void foo<T>() {}
  void bar<T extends AnotherClass> {}

  void tests() {
    foo(); // ⚠️ The type argument(s) of the function 'foo' can't be inferred.
    bar(); // No warnings
  }
}
```
